### ISSUE TITLE
Configurable std_deviation bounds

### ISSUE THREAD

**Issue by rashidkpc** [elasticsearch/elasticsearch#9356]
As we've started implementing support for the extended_stats agg in Kibana we've noticed that plotting standard deviation itself isn't useful to many people. It would be great if we could get the upper and lower bound calculated in elasticsearch

Basically looking for:

```json
{
  "std_deviation_above_avg": 40,
  "std_deviation_below_avg": 60,
  "avg": 50,
  "std_deviation": 10,
}
```

**Bonus round**: A configurable parameter for # of standard deviations away from the mean. Eg, 2, 3 std_deviations.

**Comment by spenceralger** [comment]
**Super bonus round:** if stderr is useful, could we include that too?

**Comment by polyfractal** [comment]
Standard error could also be calculated...we'll just want to put a note in the documentation that it may or may not be useful depending on your data.

E.g. if your data is an approximation of a continuous value (like sampling the CPU utilization every 30 seconds), the stderr represents how accurate your estimate of the mean is to the real CPU mean.

If your data represents a discrete set of values (like how many people drove over a specific bridge every 30 seconds), you already know the "estimate" of the mean is the exact mean, because your sample already includes *all* the data.  So in that case, the stderr should actually be zero instead of whatever we calculate.

Tl;dr: we just need to add a disclaimer to the docs.

I would also want to include confidence intervals if we add stderr.  I'm thinking syntax should be something like:

```
"extended_stats" : { 
  "field" : "price",
  "std_bounds" : "upper | lower | both",
  "std_sigmas": 2,
  "confidence_bounds" : "upper | lower | both",
  "confidence_levels" : 2
}
```

I'll work on this tomorrow, got caught up on something else today!

**Comment by simianhacker** [comment]
+1,000,000 * ∞
